crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -0,0 +1,48 @@
+use tracing::debug;
+use turbopath::{AbsoluteSystemPath, AnchoredSystemPathBuf};
+
+use crate::package_graph;
+
+pub struct PackageInference {
+    package_name: Option<String>,
+    directory_root: AnchoredSystemPathBuf,
+}
+
+impl PackageInference {
+    pub fn calculate(
```

Comment:
I kinda want a doc comment here explaining what exactly the package inference algorithm should do.